synapse/federation/transport/server/_base.py

```diff
@@ -301,7 +313,7 @@ async def new_func(request, *args, **kwargs):
                                 "client disconnected before we started processing "
                                 "request"
                             )
-                            return -1, None
+                            return None
```

Comment:
Will this change have any unwanted side effects?

I'm failing to find where the return of this function is consumed...

Reply:
Oh yeah this goes through many many layers, I should have deep linked. Sorry about that!

This essentially ends up being called at:

https://github.com/matrix-org/synapse/blob/b64b6d12d47c012a9996a2585a969d3b962d25fa/synapse/http/server.py#L262-L266

So by returning `None` we're short-circuiting calling `self._send_response`. For the case of the federation resources, they're registered as part of the [`TransportLayerServer`](https://github.com/matrix-org/synapse/blob/a6f1a3abecf8e8fd3e1bff439a06b853df18f194/synapse/federation/transport/server/__init__.py#L46) class, which is a sub-class of [`JsonResource`](https://github.com/matrix-org/synapse/blob/b64b6d12d47c012a9996a2585a969d3b962d25fa/synapse/http/server.py#L355), which is a sub-class of `DirectServeJsonResource` which essentially calls out to `response_with_json`:

https://github.com/matrix-org/synapse/blob/b64b6d12d47c012a9996a2585a969d3b962d25fa/synapse/http/server.py#L325-L339

If we investigate `response_with_json` we'll see that this also inspects `request._disconnected` and bails:

https://github.com/matrix-org/synapse/blob/b64b6d12d47c012a9996a2585a969d3b962d25fa/synapse/http/server.py#L667-L674

Pretty much, the callbacks already handle returning `None` to say "I'm done", but for some reason the federation code does something different and relies on us separate check to give up. I'm making this a bit more consistent with the C-S code. I think.

Let me know if that isn't clear!

Reply:
Thanks for clearing that up!